### PR TITLE
[IA-4833] move files to package

### PR DESCRIPTION
**Pull request by jdcanas**
In this PR, I move over the data clients and their dependencies, but I do not wire this client up to the main terra ui code base yet. There will be two follow-on PRs in this order:
- Wire up the `Disks` auth injection wrapper with new package files and migrate the tests
- Properly move over the `data-client-core` additions `FetchFn` and `AbortOptions`

### CODE REVIEW

File: packages/data-clients/leonardo/src/LeoDisksDataClient.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }),
+  });
+
+  return {
+    disksV1: v1Func,
+    disksV2: () => v2Api,
+  };
+};
+
+export type DisksHelperDeps = {
```
Comment (msilva-broad) [inline packages/data-clients/leonardo/src/LeoDisksDataClient.ts]: lets move this up near the top of the file before it's used in the function signature of makeDisksHelper

File: .yarn/cache/@types-qs-npm-6.9.15-aae1b1e2f7-97d8208c2b.zip
Comment (LizBaldo) [inline .yarn/cache/@types-qs-npm-6.9.15-aae1b1e2f7-97d8208c2b.zip:1]: I am just curious if these files should be committed or not?

File: packages/data-clients/leonardo/src/LeoDisksV2DataClient.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { AbortOption, FetchFn } from '@terra-ui-packages/data-client-core';
+
+export interface LeoDisksV2DataClient {
+  delete: (diskId: number, options?: AbortOption) => Promise<void>;
+}
+const diskV2Root = 'api/v2/disks';
+
+export const makeLeoDisksV2DataClient = (deps: LeoDisksV2DataClientDeps): LeoDisksV2DataClient => {
```
Comment (LizBaldo) [inline packages/data-clients/leonardo/src/LeoDisksV2DataClient.ts:8]: This is definitely not part of this PR, but I am still a bit confused at the huge difference between the use of the V1 vs V2 disk API in the UI, don't we need to list V2 disks as well? Or is the V1 list combining V1 and V2 resources already?

**Review by LizBaldo — APPROVED** [review]
I only have a couple of super minor comments that should not reauire code changes, LGTM 👍

**Review by msilva-broad — APPROVED** [review]
awesome work!

File: .yarn/cache/@types-qs-npm-6.9.15-aae1b1e2f7-97d8208c2b.zip
Comment (jdcanas) [reply .yarn/cache/@types-qs-npm-6.9.15-aae1b1e2f7-97d8208c2b.zip:1]: yes, there's actually a precommit hook to ensure the yarn cache is committed. This ensures reproducible installs.

File: packages/data-clients/leonardo/src/LeoDisksV2DataClient.ts (hunk 2 above)
Comment (jdcanas) [reply packages/data-clients/leonardo/src/LeoDisksV2DataClient.ts:8]: we don't manage v2 disks directly beyond deleting, all other operations are done in the context of a VM (i.e. creating a disk with a VM). 

The list endpoint already lists all disks, since there's an option to not pass in googleProject as a filter
